apollo/src/graphql/storage-provider/storageProviderQueries.ts

```diff
@@ -27,4 +30,96 @@ builder.queryFields((t) => ({
             return storageProvider;
         },
     }),
+    listObjectsForModelVersion: t.field({
+        type: StorageProviderObjectConnection,
+        authScopes: {
+            loggedIn: true,
+        },
+        args: {
+            modelVersionId: t.arg.string({ required: true }),
+            // TODO: Putting defaultValue & required overrides defaultValue
+            first: t.arg({
+                type: 'Limit',
+                defaultValue: 10,
+                required: true,
+            }),
+            after: t.arg.string(),
+            prefix: t.arg.string(),
+        },
+        async resolve(_root, args, _ctx) {
+            const splitKey = (prefixLength: number, objectKey?: string): string[] => {
+                if (objectKey) {
+                    return objectKey.slice(prefixLength + 1).split('/');
+                }
+
+                return [];
+            };
+
+            const folderOrFile = (keyArr?: string[]): 'folder' | 'file' | undefined => {
+                if (keyArr) {
+                    return keyArr.length > 1 ? 'folder' : 'file';
+                }
+
+                return undefined;
+            };
+
+            const modelStorageProvider = (await ObjectionMLModelVersion.relatedQuery(
```

Comment:
We need to make sure that the user has permission to view the model object

Reply:
Noted.

Reply:
Wait, brain fart. What criteria do we impose on users such that they have permissions to view the model objects? I just have the `loggedIn: true` set for the `authScopes`.

Reply:
So that Pothos scope just validates that they're using a valid auth token. We still need to query what team the model belongs to, and that the authenticated user is aember of that team. I think the last PR or two I submitted do these checks if you want to crib that code, it's all pretty copy paste until we get aggravated and build a better abstraction

Reply:
Gotcha, yeah it's #119